airflow/utils/log/stackdriver_task_handler.py

```diff
@@ -15,281 +15,14 @@
-    ) -> Tuple[str, bool, Optional[str]]:
-        """
-        Sends requests to the Stackdriver service and downloads logs.
-
-        :param log_filter: Filter specifying the logs to be downloaded.
-        :type log_filter: str
-        :param next_page_token: The token of the page from which the log download will start.
-            If None is passed, it will start from the first page.
-        :param all_pages: If True is passed, all subpages will be downloaded. Otherwise, only the first
-            page will be downloaded
-        :return: A token that contains the following items:
-            * string with logs
-            * Boolean value describing whether there are more logs,
-            * token of the next page
-        :rtype: Tuple[str, bool, str]
-        """
-        messages = []
-        new_messages, next_page_token = self._read_single_logs_page(
-            log_filter=log_filter,
-            page_token=next_page_token,
-        )
-        messages.append(new_messages)
-        if all_pages:
-            while next_page_token:
-                new_messages, next_page_token = self._read_single_logs_page(
-                    log_filter=log_filter,
-                    page_token=next_page_token
-                )
-                messages.append(new_messages)
-
-            end_of_log = True
-            next_page_token = None
-        else:
-            end_of_log = not bool(next_page_token)
-        return "\n".join(messages), end_of_log, next_page_token
-
-    def _read_single_logs_page(self, log_filter: str, page_token: Optional[str] = None) -> Tuple[str, str]:
-        """
-        Sends requests to the Stackdriver service and downloads single pages with logs.
-
-        :param log_filter: Filter specifying the logs to be downloaded.
-        :type log_filter: str
-        :param page_token: The token of the page to be downloaded. If None is passed, the first page will be
-            downloaded.
-        :type page_token: str
-        :return: Downloaded logs and next page token
-        :rtype: Tuple[str, str]
-        """
-        entries = self._client.list_entries(filter_=log_filter, page_token=page_token)
-        page = next(entries.pages)
-        next_page_token = entries.next_page_token
-        messages = []
-        for entry in page:
-            if "message" in entry.payload:
-                messages.append(entry.payload["message"])
-
-        return "\n".join(messages), next_page_token
-
-    @classmethod
-    def _task_instance_to_labels(cls, ti: TaskInstance) -> Dict[str, str]:
-        return {
-            cls.LABEL_TASK_ID: ti.task_id,
-            cls.LABEL_DAG_ID: ti.dag_id,
-            cls.LABEL_EXECUTION_DATE: str(ti.execution_date.isoformat()),
-            cls.LABEL_TRY_NUMBER: str(ti.try_number),
-        }
+warnings.warn(
+    "This module is deprecated. Please use `airflow.providers.google.cloud.log.stackdriver_task_handler`.",
+    DeprecationWarning, stacklevel=2
+)
```

Comment:
Should we add test in `test_core_to_contrib.py`?

Reply:
Wow. Have been thinking of what to test and how this should be tested. Thanks. Will do it now